From a5a07225bd98a6c8b30625008cb08ef3616f1a44 Mon Sep 17 00:00:00 2001
From: Alexander Bernauer <alex@copton.net>
Date: Fri, 16 Mar 2012 12:23:07 +0100
Subject: [PATCH] detailed CPU cycle tracking for processes

---
 core/sys/process.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/core/sys/process.c b/core/sys/process.c
index 70adf67..aea0c03 100644
--- a/core/sys/process.c
+++ b/core/sys/process.c
@@ -243,6 +243,7 @@ do_poll(void)
  * listening processes.
  */
 /*---------------------------------------------------------------------------*/
+volatile void* process_hook = 0;
 static void
 do_event(void)
 {
@@ -282,7 +283,9 @@ do_event(void)
 	if(poll_requested) {
 	  do_poll();
 	}
+    process_hook = p;
 	call_process(p, ev, data);
+    process_hook = 0;
       }
     } else {
       /* This is not a broadcast event, so we deliver it to the
-- 
1.7.1

